pkgs/by-name/li/libfprint-focaltech-2808-a658/package.nix

```diff
@@ -0,0 +1,106 @@
+{
+  stdenv,
+  lib,
+  fetchurl,
+  rpm,
+  cpio,
+  glib,
+  gusb,
+  pixman,
+  libgudev,
+  nss,
+  libfprint,
+  cairo,
+  pkg-config,
+  autoPatchelfHook,
+  makePkgconfigItem,
+  copyPkgconfigItems,
+}:
+
+# https://discourse.nixos.org/t/request-for-libfprint-port-for-2808-a658/55474
+let
+  pname = "libfprint-focaltech-2808-a658";
+  version = "1.94.4";
+  # library name in source
+  libso = "libfprint-2.so.2.0.0";
+in
+stdenv.mkDerivation rec {
+  inherit pname;
+  inherit version;
+  # https://gitlab.freedesktop.org/libfprint/libfprint/-/merge_requests/413#note_2476573
+  src = fetchurl {
+    url = "https://github.com/ftfpteams/RTS5811-FT9366-fingerprint-linux-driver-with-VID-2808-and-PID-a658/raw/b040ccd953c27e26c1285c456b4264e70b36bc3f/libfprint-2-2-${version}+tod1-FT9366_20240627.x86_64.rpm";
+    hash = "sha256-MRWHwBievAfTfQqjs1WGKBnht9cIDj9aYiT3YJ0/CUM=";
+  };
+
+  nativeBuildInputs = [
+    rpm
+    cpio
+    pkg-config
+    autoPatchelfHook
+    copyPkgconfigItems
+  ];
+
+  buildInputs = [
+    stdenv.cc.cc
+    glib
+    gusb
+    pixman
+    nss
+    libgudev
+    libfprint
+    cairo
+  ];
+
+  unpackPhase = ''
+    runHook preUnpack
+
+    rpm2cpio $src | cpio -idmv
+
+    runHook postUnpack
+  '';
+
+  # custom pkg-config based on libfprint's pkg-config
+  pkgconfigItems = [
+    (makePkgconfigItem rec {
+      name = "libfprint-2";
+      inherit version;
+      inherit (meta) description;
+      cflags = [ "-I${variables.includedir}/libfprint-2" ];
+      libs = [
+        "-L${variables.libdir}"
+        "-lfprint-2"
+      ];
+      variables = rec {
+        prefix = "${placeholder "out"}";
```

Comment:
So this just... works in `makePkgconfigItem`? What wizardry is this?

Reply:
Yeah, referred it from https://github.com/NixOS/nixpkgs/blob/8bba3c2d4607ae405deb1bff6c8247bfd48454fd/pkgs/development/compilers/tinycc/default.nix#L32-L51 and https://github.com/NixOS/nixpkgs/blob/8bba3c2d4607ae405deb1bff6c8247bfd48454fd/pkgs/development/libraries/stb/default.nix#L16-L27

This makes our custom pkg-config file, just like we wanted.
Yes, this just works!